src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -9831,27 +9831,27 @@ String StorageReplicatedMergeTree::findReplicaHavingPart(
     return {};
 }
 
-
 bool StorageReplicatedMergeTree::checkIfDetachedPartExists(const String & part_name)
 {
-    fs::directory_iterator dir_end;
-    for (const std::string & path : getDataPaths())
-        for (fs::directory_iterator dir_it{fs::path(path) / "detached/"}; dir_it != dir_end; ++dir_it)
-            if (dir_it->path().filename().string() == part_name)
+    for (const auto & disk : getStoragePolicy()->getDisks())
+    {
```

Comment:
There is still no `disk->isBroken()` check.

Reply:
I think `iterateDirectory` should return empty iterator if disk is broken. At least it's true for disk local but I haven't check other type (though I think this PR mainly target disk local).
@amosbird do you think we still need to explicitly check if disk is broken?

Reply:
Oh, I see.